subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -282,7 +282,9 @@ void fp_keys_bt_auth_progress(const struct bt_conn *conn, bool authenticated)
 	struct fp_procedure *proc = &fp_procedures[bt_conn_index(conn)];
 
 	if (proc->state == FP_STATE_USE_TEMP_KEY) {
-		(void)k_work_reschedule(&proc->timeout, FP_KEY_TIMEOUT);
+		(void)k_work_cancel_delayable(&proc->timeout);
+		(void)k_work_schedule(&proc->timeout, FP_KEY_TIMEOUT);
```

Comment:
We are relaying on the fact that we are in cooperative thread and the work cannot be in RUN state, it means that cancel should always work - it is implementation detail, and such "*should*" needs assert test.